### PR TITLE
added database support for the bot

### PR DESCRIPTION
**Pull request by Prajita99**

### CODE REVIEW

File: mantra/config/__init__.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
 from .bot_config import bot_config
+from .database import db_config
 
 __all__ = ["bot_config"]
```
Comment (thenishantsapkota) [inline mantra/config/__init__.py]: Add `db_config` to the `__all__` variable

File: mantra/config/database.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from pydantic import BaseSettings
+
+
+class PostgresConfig(BaseSettings):
+    user: str
+    db: str
+    password: str
+    port: str
```
Comment (thenishantsapkota) [inline mantra/config/database.py]: Port number should be `int` not `str`.

File: mantra/core/bot.py [hunk 3]
```diff
@@ -29,6 +33,7 @@ def run_bot(self) -> None:
         super().run(asyncio_debug=True)
 
     async def on_starting(self, event: hikari.StartingEvent) -> None:
+        asyncio.create_task(self.establish_db_connection())
```
Comment (thenishantsapkota) [inline mantra/core/bot.py:36]: Just like establishing a connection, it is equally important to close the connections when exiting the bot.
Please add the functionality for that

File: mantra/core/bot.py [hunk 4]
```diff
@@ -41,3 +46,6 @@ async def on_stopping(self, event: hikari.StoppingEvent) -> None:
 
     async def on_stopped(self, event: hikari.StoppedEvent) -> None:
         ...
+
+    async def establish_db_connection(self):
```
Comment (thenishantsapkota) [inline mantra/core/bot.py]: Specify the return type as `None` here

File: migrations/main/0_20221007103141_init.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+from tortoise import BaseDBAsyncClient
```
Comment (thenishantsapkota) [inline migrations/main/0_20221007103141_init.py]: This shouldn't be commited to GitHub.
Please remove it.

**Review by thenishantsapkota — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: migrations/main/0_20221007103141_init.py (hunk 5 above)
Comment (thenishantsapkota) [reply migrations/main/0_20221007103141_init.py]: My bad this has already been deleted!

File: mantra/config/__init__.py (hunk 1 above)
Comment (Prajita99) [reply mantra/config/__init__.py]: Will do!

File: mantra/config/database.py (hunk 2 above)
Comment (Prajita99) [reply mantra/config/database.py]: I didn't think of that 
It'll be fixed

**Comment by thenishantsapkota** [comment]
Please also update `.env.example` because this PR introduces some new environment vars.

**Review by thenishantsapkota — APPROVED** [review]

**Pull request merged**